File: kid-problem-ui-test-selenium/KpUiTestxUnit/Models/ExamRun.cs
namespace KpUiTestxUnit.Models;

public class ExamRun
{
    public string? UID { get; set; }
    public string? ExamCategory { get; set; }
    public string? ExamTitle { get; set; }
    public string? AnsweredBy { get; set; }
    public string? StartTime { get; set; }
    public string? CompleteTime { get; set; }
    public string? TotalDuration { get; set; }
    public int? TotalCount { get; set; }
    public int? CorrectCount { get; set; }
    public int? GuessCount { get; set; }
    public int? GuessCorrectCount { get; set; }

    public ExamRunDetail[]? ExamDetails { get; set; }

}

File: kidproblem-client/src/app/_models/exam-run.ts
import { BaseModel } from '@app/_models/adm/base-model';

export class ExamRun extends BaseModel {
    Id: string = '';
    AnswerBy: string = '';
    AnswerByFullname: string = '';
    CreateTime!: Date;
    AssignmentId: string = '';
    ExamTitle: string = '';
    ExamCategory: string = '';
    StartTime?: Date;
    CompleteTime?: Date;
    FamilyId?: string;
    TotalCount: number = 0;
    CorrectCount?: number;
    GuessCount?: number;
    GuessCorrectCount?: number;
    TotalDuration?: number;
    ExamRunDetails: ExamRunDetail[] = [];
    Amc10Score?: number;
}

export interface ExamRunDetail {
    Id: string;
    ProblemTitle: string;
    UserAnswer?: string;
    IsGuess?: boolean;
    Duration?: number;
    IsCorrect?: boolean;
}